app.js

```diff
@@ -23,15 +24,34 @@ app.all('/*', (req, res) => {
 app.use((err, req, res, next) => {
     if(err.status && err.msg){
         res.status(err.status).send({msg: err.msg})
+    } 
+    else {
+        next(err);
     }
-    else if (err.code === '22P02'){
-       res.status(400)
-       res.send({msg: 'Bad Request'}) 
-    }
-    else if (!err) {
+});
+
+app.use((err, req, res, next) => {
+    if (err.code === '22P02'){
+        res.status(400)
+        res.send({msg: 'Bad Request'}) 
+     }
+     else {
+         next(err);
+     }
+});
+
+app.use((err, req, res, next) => {
+    if(err.code === '23503'){
         res.status(404)
-        res.send({msg: 'Not Found'})
+        if(err.constraint === 'comments_author_fkey'){
+            res.send({msg: 'No Such User'})
+        }
```

Comment:
This is a nice touch but it may become inaccurate if there's a different foreign key causing the problem